Snowflake.Data/Core/FileTransfer/StorageClient/SFS3Client.cs

```diff
@@ -309,23 +309,25 @@ private FileHeader HandleFileHeaderResponse(ref SFFileMetadata fileMetadata, Get
             }
 
             // If a specific endpoint is specified use this
-            if ((null != endpoint) && (0 != endpoint.Length))
+            if (!string.IsNullOrEmpty(endpoint))
             {
                 var start = endpoint.IndexOf('[');
                 var end = endpoint.IndexOf(']');
-                if(start > -1 && end > -1 && end > start)
+                if (start > -1 && end > -1 && end > start)
                 {
                     endpoint = endpoint.Substring(start + 1, end - start - 1);
-                    if(!endpoint.Contains("https"))
-                    {
-                        endpoint = "https://" + endpoint;
-                    }
                 }
+
+                if (!endpoint.Contains("https://"))
```

Comment:
what do you think about `StartsWith("https://", StringComparison.OrdinalIgnoreCase)` ?

Reply:
fixed in [9bb006b](https://github.com/snowflakedb/snowflake-connector-net/pull/914/commits/9bb006b2ba4c6437e4a7165b34ba9aed6cdcc5fc)